runtime/doc/quickfix.txt

```diff
@@ -1335,6 +1335,59 @@ g:javac_makeprg_params variable.  For example: >
 
 	let g:javac_makeprg_params = "-Xlint:all -encoding utf-8"
 <
+SPOTBUGS						*compiler-spotbugs*
+
+SpotBugs is a static analysis tool that can be used to find bugs in Java.
+It scans the Java bytecode of all classes in the currently open buffer.
+
+Commonly used compiler options can be added to 'makeprg' by setting the
+b/g:spotbugs_makeprg_params variable.  For example: >
+
+	let b:spotbugs_makeprg_params = "-longBugCodes -effort:max -low"
+
+The global default is "-workHard -experimental".
+
+By default, the class files are searched in the directory where the source
+files are placed.  However, typical Java projects use distinct directories
+for source files and class files.  To make both known to SpotBugs, assign
+their paths (relative to the current directory) to the following two
```

Comment:
The ‘**relative to the current directory**’ part is misleading!

Let me draw a few pictures to clarify what I mean:

(Generic)

```
|- common --| |- distinct -| |- ~ common -|
/tmp/project/ src/           com/acme/Foo.*

|- common --| |- distinct -| |- ~ common -|
/tmp/project/ bin/           com/acme/Foo.*
```

(Maven)

```
|- common --| |- distinct -|       |- ~ common -|
/tmp/project/ src/main/java/       com/acme/Foo.*

|- common --| |---- distinct ----| |- ~ common -|
/tmp/project/ target/classes/      com/acme/Foo.*

```

(Modular Ant)

```
|- common --| |------- distinct ------|   |- ~ common -|
/tmp/project/ src/org.acme.api/classes/   com/acme/Foo.*

|- common --| |------- distinct --------| |- ~ common -|
/tmp/project/ build/modules/org.acme.api/ com/acme/Foo.*
```

For the class-file-look-up algorithm to succeed, we need to  
obtain the distinct pairs from `g:spotbugs_properties` so that  
we can swap out the part _and_ swap out the file extension  
before we test a produced candidate name (along with `\$*`  
candidate names) with `filereadable()`.  The current working  
directory is irrelevant for the algorithm and, by extension,  
for SpotBugs since the absolute path is used; but it can be  
relevant for Javac's `-d` and `-cp` flags should those use  
relative paths.  The distinct paths are relative to _each  
other_ not the current working directory of a Vim instance.

Reply:
> 'relative to the current directory’ part is misleading

Fully agreed; I meant common root directory. Does that make things better?

Reply:
We, rather I, have to be careful about the assumption that Vim was started in the project root dir for things to work